### PR TITLE
Alpha/build question function

### PR DESCRIPTION
**Pull request by shaargtz**
# What does this PR do?
- Makes the buildQuestion available from the Utility class to be reused.

**Related Issues**
- NA

---
**Merge Checklist**
- [x] Branch is up to date
- [x] No code style issues

### CODE REVIEW

File: src/main/java/com/google/sps/classes/Utility.java [hunk 1]
```diff
@@ -17,9 +17,10 @@
 import com.google.appengine.api.users.UserService;
 import com.google.appengine.api.users.UserServiceFactory;
 import com.google.gson.Gson;
+import com.google.sps.classes.SqlConstants;
 import java.sql.*;
 import java.util.logging.Level;
-import java.util.logging.Logger;
+import java.util.logging.Logger;
```
Comment (OUMontiel) [inline src/main/java/com/google/sps/classes/Utility.java:23]: nit: Remove space at end of line.

File: src/main/java/com/google/sps/classes/Utility.java [hunk 2]
```diff
@@ -32,16 +33,19 @@
...
       + "LEFT JOIN (SELECT question_id, COUNT(follower_id) followers FROM QuestionFollower "
-      + "GROUP BY question_id) CountTable ON Question.id=CountTable.question_id "
-      + "LEFT JOIN (SELECT username, id AS asker_id FROM User) NameTable "
-      + "ON Question.asker_id=NameTable.asker_id "
+      + "GROUP BY question_id) FollowerCount ON Question.id=FollowerCount.question_id "
+      + "LEFT JOIN (SELECT username, id AS asker_id FROM User) GetUsername "
+      + "ON Question.asker_id=GetUsername.asker_id "
       + "LEFT JOIN (SELECT question_id, COUNT(id) answers FROM Answer "
-      + "GROUP BY question_id) AnswerTable ON Question.id=AnswerTable.question_id ";
+      + "GROUP BY question_id) AnswerCount ON Question.id=AnswerCount.question_id ";
```
Comment (OUMontiel) [inline src/main/java/com/google/sps/classes/Utility.java:48]: I see that the question's ID appears three times in this table. Is it really needed? If not, maybe removing `question_id` from lines 43 and 47 works?

File: src/main/java/com/google/sps/classes/Utility.java [hunk 3]
```diff
@@ -125,4 +129,28 @@ public static void addNewUser(String firstName, String lastName, String username
       logger.log(Level.SEVERE, exception.getMessage(), exception);
     }
   }
+
+  /** 
+   * Create a question object using the results from a query.
+   */
+  public static QuestionObject buildQuestion(ResultSet queryResult) {
```
Comment (OUMontiel) [inline src/main/java/com/google/sps/classes/Utility.java:136]: I know it doesn't have to do specifically with this PR, but do you think changing the name of `QuestionObject` to just `Question` would be a better idea? I think the `Object` is redundant.

**Comment by shaargtz** [comment]
this is no longer going to be useful

File: src/main/java/com/google/sps/classes/Utility.java (hunk 1 above)
Comment (shaargtz) [reply src/main/java/com/google/sps/classes/Utility.java:23]: done

File: src/main/java/com/google/sps/classes/Utility.java (hunk 2 above)
Comment (shaargtz) [reply src/main/java/com/google/sps/classes/Utility.java:48]: but those IDs are needed to perform the joins, i don't know of any other way to do it

File: src/main/java/com/google/sps/classes/Utility.java (hunk 3 above)
Comment (shaargtz) [reply src/main/java/com/google/sps/classes/Utility.java:136]: yep, I did it in another branch, thanks!

File: src/main/java/com/google/sps/classes/Utility.java (hunk 2 above)
Comment (OUMontiel) [reply src/main/java/com/google/sps/classes/Utility.java]: Ooh, okok, got it. It's all good then!

File: src/main/java/com/google/sps/classes/Utility.java [hunk 4]
```diff
@@ -34,16 +35,19 @@
...
   
-  // Query to retrieve data from a question. The ? at the end must be replaced in the
-  // prepared statement, can be '1=1' for all questions or a different condition to match
-  // the questions that are needed.
+  // Query to retrieve data from a question. Generates the following table:
+  //
+  // |-----------------Question-----------------|------FollowerCount------|-----GetUsername-----|------AnswerCount------|
+  // +----+-------+------+----------+-----------+-------------+-----------+----------+----------+-------------+---------+
+  // | id | title | body | asker_id | date_time | question_id | followers | username | asker_id | question_id | answers |
+  // +----+-------+------+----------+-----------+-------------+-----------+----------+----------+-------------+---------+
```
Comment (gloriapark) [inline src/main/java/com/google/sps/classes/Utility.java:43]: This is great!!

**Pull request merged**